smallrye-reactive-messaging-aws-sns-1.0/src/main/java/io/smallrye/reactive/messaging/aws/sns/SnsVerticle.java

```diff
@@ -0,0 +1,128 @@
+package io.smallrye.reactive.messaging.aws.sns;
+
+import java.io.ByteArrayInputStream;
+import java.util.Optional;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+
+import com.amazonaws.services.sns.message.DefaultSnsMessageHandler;
+import com.amazonaws.services.sns.message.SnsMessageManager;
+import com.amazonaws.services.sns.message.SnsNotification;
+
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BodyHandler;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.apache.ApacheSdkHttpService;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.CreateTopicRequest;
+import software.amazon.awssdk.services.sns.model.CreateTopicResponse;
+import software.amazon.awssdk.services.sns.model.ListSubscriptionsByTopicRequest;
+import software.amazon.awssdk.services.sns.model.ListSubscriptionsByTopicResponse;
+import software.amazon.awssdk.services.sns.model.SubscribeRequest;
+import software.amazon.awssdk.services.sns.model.SubscribeResponse;
+import software.amazon.awssdk.services.sns.model.Subscription;
+
+public class SnsVerticle extends AbstractVerticle {
+
+    private final String topicName;
+    private final String endpoint;
+    private final int port;
+    private static final Logger LOG = LoggerFactory.getLogger(SnsVerticle.class);
+    private SnsMessageManager messageManager = new SnsMessageManager();
+    private BlockingQueue<SnsMessage<String>> msgQ = new LinkedBlockingDeque<>();
+
+    public SnsVerticle(String endpoint, String topicName, int port) {
+        this.topicName = topicName;
+        this.endpoint = endpoint;
+        this.port = port;
+    }
+
+    @Override
+    public void start(Future<Void> startFuture) {
+
+        Router router = Router.router(vertx);
+        router.route()
+                .handler(BodyHandler.create())
```

Comment:
should be written as:

```
router.post().handler(BodyHandler.create());
```

Reply:
I didn't handle body using body handler because I used Amazon SDK SnsMessageManager in  receiveSnsMsg to handle message body. it automatically do AWS topic subscription confirmation and format JSON message into SnsMessage.